storage/innobase/trx/trx0rseg.cc

```diff
@@ -101,9 +111,15 @@ trx_rseg_update_wsrep_checkpoint(
 }
 
 /** Update WSREP checkpoint XID in first rollback segment header.
+This function should be called only from innodbase_wsrep_set_checkopoint()
+when it is guaranteed that there are no wsrep transactions committing.
+If the UUID part of the wsrep XID does not match to the UUIDs of XIDs already
+stored into rollback segments, all the remaining XIDs will be reset to empty
+value.
 @param[in]	xid		WSREP XID */
 void trx_rseg_update_wsrep_checkpoint(const XID* xid)
 {
+	static byte current_xid_uuid[16] = {0, };
```

Comment:
I would prefer a stack-allocated buffer that would be initialized by `memset()`.

Reply:
The idea of static variable is to maintain the value over calls to this function. As mentioned above, this function is called for TOI and membership view events only, so it is guaranteed that only single thread is accessing this function at the time (with suitable memory barriers).